src/executor/graph_executor.cc

```diff
@@ -444,6 +445,15 @@ void GraphExecutor::InitCachedOps() {
   for (uint32_t nid = 0; nid < idx.num_nodes(); ++nid) {
     const auto& inode = idx[nid];
     if (inode.source->is_variable()) continue;
+#if MXNET_USE_PROFILER
+    // TODO(ziheng) memory leak risk
+    int len = inode.source->attrs.name.size();
```

Comment:
This is a memleak point. Sorry I didn't think it through when I asked you to change opr_name to const char*.
The problem is dynamic message vs static message. 
If we want dynamic message, where we can generate different messages, then opr_name should be string. This is more flexible but slower.
If we want static message, where every line of push can only have the same message, ie a const char*, then opr_name should be char*.

After all the speed difference probably wouldn't be that big. You can change it back to string if you think that's better. Just make sure to use std::move when possible to make sure there aren't repeated constructions.